.github/workflows/main.yml

```diff
@@ -0,0 +1,72 @@
+name: Pipeline for Infrastructure provisioning and configuration
+
+run-name: ${{ github.actor }} is testing the pipeline
+
+on:
```

Comment:
why workflow_dispatch here ?
and push on branches

Reply:
workflow_dispatch is kept as a trigger so that after merging the branch on the main branch, we can run the workflow manually from the github UI. Likewise, the push is on branches so that the workflow is triggered when we push to the INTERN-761 branch. I have used square brackets so that I can add other branches like the main later if I want.